apps/aecore/src/aec_trees.erl

```diff
@@ -29,12 +30,17 @@
 
 -spec new() -> trees().
 new() ->
-    #trees{accounts  = aec_accounts_trees:empty(),
-           oracles   = aeo_state_tree:empty(),
-           contracts = aect_state_tree:empty(),
+    #trees{accounts  = aec_accounts_trees:empty_with_backend(),
+           oracles   = aeo_state_tree:empty_with_backend(),
+           contracts = aect_state_tree:empty_with_backend(),
            ns        = aens_state_tree:empty()
           }.
 
+-spec commit_to_db(trees()) -> trees().
+commit_to_db(Trees) ->
+    %% Make this in a transaction to get atomicity.
```

Comment:
Do you mean DB transaction between state trees and block? (in aec_chain_state:state_db_put)

Reply:
I am referring to the line below. The commit of the state trees need to be atomic to not miss any nodes.